### PR TITLE
Make bxl self-host use internal symbol publishing

### PR DESCRIPTION
**Pull request by olkononenko**
[AB#1607767](https://dev.azure.com/mseng/708e929f-6bd5-415a-8daf-25b1dac08dd8/_workitems/edit/1607767)

### CODE REVIEW

File: Public/Src/Tools/SymbolDaemon/Program.cs [hunk 1]
```diff
@@ -26,12 +32,38 @@ public static int Main(string[] args)
...
+                Console.WriteLine(SymbolDaemon.SymbolDLogPrefix + string.Join(Environment.NewLine + SymbolDaemon.SymbolDLogPrefix, args));
                 Console.WriteLine();
 
                 SymbolDaemon.EnsureCommandsInitialized();
 
-                var confCommand = ServicePipDaemon.ServicePipDaemon.ParseArgs(args, new UnixParser());
+                // resolve response files
+                var resolvedArguments = args.SelectMany(arg =>
+                    {
+                        if (arg.StartsWith(ResponseFilePrefix))
```
Comment (narasamdya) [inline Public/Src/Tools/SymbolDaemon/Program.cs]: >ResponseFilePrefix [](start = 43, length = 18)

We have a logic for handling response file. See [http://index/?leftProject=BuildXL.ToolSupport&leftSymbol=gwduauxhomsh&file=CommandLineUtilities.cs&line=649](http://index/?leftProject=BuildXL.ToolSupport&leftSymbol=gwduauxhomsh&file=CommandLineUtilities.cs&line=649)

File: Public/Src/Tools/SymbolDaemon/SymbolDaemon.cs [hunk 2]
```diff
@@ -321,14 +321,21 @@ public static void SerializeSymbolsMetadata(Dictionary<ContentHash, HashSet<Debu
                 for (int i = 0; i < hashCount; i++)
                 {
                     ContentHash.TryParse(reader.ReadLine(), out var hash);
+                    Contract.Assert(hash.HashType == HashType.Vso0);
+                    var blobIdentifier = new Microsoft.VisualStudio.Services.BlobStore.Common.BlobIdentifier(hash.ToHashByteArray());
```
Comment (narasamdya) [inline Public/Src/Tools/SymbolDaemon/SymbolDaemon.cs]: >Microsoft.VisualStudio.Services.BlobStore.Common. [](start = 45, length = 49)

Can it not be shortened by `using Microsoft.VisualStudio.Services.BlobStore.Common` ?

Comment (olkononenko) [reply Public/Src/Tools/SymbolDaemon/SymbolDaemon.cs]: no - the same class is defined in BuildXL.Cache.* and in Microsoft.VisualStudio.*; so I need a full name

---
In reply to: [336120401](https://github.com/microsoft/BuildXL/pull/1041#discussion_r336120401) [](ancestors = 336120401)

File: Public/Src/Tools/SymbolDaemon/Program.cs (hunk 1 above)
Comment (olkononenko) [reply Public/Src/Tools/SymbolDaemon/Program.cs]: discussed offline

---
In reply to: [336119672](https://github.com/microsoft/BuildXL/pull/1041#discussion_r336119672) [](ancestors = 336119672)

**Pull request merged**